### PR TITLE
feature(toggle-shell): use keyboard to toggle mongosh COMPASS-5385

### PR DESCRIPTION
**Pull request by mabaasit**
feat(shell-toggle): enable keyboard toggle for shell COMPASS-5385

### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added

## Motivation and Context

- [ ] Bugfix
- [x] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [x] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass-shell/src/components/compass-shell/compass-shell.jsx [hunk 1]
```diff
@@ -136,6 +138,13 @@ export class CompassShell extends Component {
     );
   }
 
+  hideInfoModal() {
+    this.setState({ showInfoModal: false });
+    if (this.shellRef.current) {
```
Comment (lerouxb) [inline packages/compass-shell/src/components/compass-shell/compass-shell.jsx:143]: Is this check necessary? How would you get here if current is falsey?

**Review by lerouxb — APPROVED** [review]

File: packages/compass-shell/src/components/compass-shell/compass-shell.jsx (hunk 1 above)
Comment (mabaasit) [reply packages/compass-shell/src/components/compass-shell/compass-shell.jsx:143]: the current can be null until the `Shell` component has rendered.

File: packages/compass-shell/src/components/shell-header/shell-header.jsx [hunk 2]
```diff
@@ -15,6 +15,20 @@ export class ShellHeader extends Component {
...
 
+  componentDidMount() {
+    document.addEventListener('keydown', this.handleKeyboardToggle.bind(this));
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keydown', this.handleKeyboardToggle.bind(this));
+  }
+
+  handleKeyboardToggle({ ctrlKey, keyCode }) {
```
Comment (gribnoysup) [inline packages/compass-shell/src/components/shell-header/shell-header.jsx:26]: Just FYI `keyCode` [is deprecated](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode)

Comment (mabaasit) [reply packages/compass-shell/src/components/shell-header/shell-header.jsx:26]: I'll create a follow up PR

Comment (mabaasit) [reply packages/compass-shell/src/components/shell-header/shell-header.jsx:26]: [Fixed here](https://github.com/mongodb-js/compass/pull/2660)

**Pull request merged**